Added documentation for Salesforce Analytics 3.2 (#819)
mulesoft/docs-connectors: salesforce-analytics-cloud/3.2/modules/ROOT/pages/salesforce-analytics-connector-reference.adoc

= Salesforce Einstein Analytics Connector Reference - Mule 4
:page-aliases: connectors::salesforce/salesforce-analytics-connector-reference.adoc

Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#anypoint-connectors[Select]

Salesforce Analytics Connector v3.2

Release Notes: xref:release-notes::connector/salesforce-analytics-connector-release-notes-mule-4.adoc[Salesforce Einstein Analytics Connector Release Notes]

== Default Configuration

=== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
|Name | String | The name for this configuration. Connectors reference the configuration with this name. | |x
| Connection a| * Basic Username and Password
* OAuth v2.0
* OAuth JWT
* OAuth 2.0 SAML Bearer

>  | The connection types that can be provided to this configuration. | |x

Reviewer: ```suggestion
 | The connection types to provide to this configuration. | |x
```

| Password a| String |  Password used to authenticate against the proxy. |  |
|===

==== Associated Operations

* createDataSet
* deleteDataSet
* startDataProcessing
* unauthorize
* uploadExternalData
* uploadExternalDataIntoNewDataSetAndStartProcessing

== Operations

[[createDataSet]]
== Create Data Set
`<salesforce-analytics:create-data-set>`

Creates a new dataset in the Salesforce Analytics Cloud system and returns the identifier of the created data set within the Salesforce Analytics Cloud system.

=== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | |x
| Type a| String |  Represents the type of the dataset to be created |  |x
| Operation a| Enumeration, one of:

** APPEND
** OVERWRITE
** UPSERT
** DELETE |  |  |x
| Description a| String |  |  |x
| Label a| String |  |  |x
| Data Set Name a| String |  |  |x
| Edgemart Container a| String |  |  |
| Notification Sent a| Enumeration, one of:

** ALWAYS
** FAILURES
** NEVER
** WARNINGS
|  |  |
| Notification Email a| String |  |  |

> | Target Variable a| String |  The variable name where the output of the operation is stored. |  |

Reviewer: ```suggestion
| Target Variable a| String |  The variable name where to store the operation's output. |  |
```

* SALESFORCE-ANALYTICS:TRANSACTION
* SALESFORCE-ANALYTICS:UNKNOWN

[[unauthorize]]
== Unauthorize

`<salesforce-analytics:unauthorize>`

Deletes all the access token information of a given resource owner ID so that it's impossible to execute any operation for the user without repeating the authorization dance.

=== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use | |x
| Resource Owner Id a| String |  The ID of the resource owner for whom to invalidate access |  |
|===

=== For Configurations

* salesforce-analytics

[[uploadExternalData]]
== Upload External Data
`<salesforce-analytics:upload-external-data>`

Inserts records into a dataset at an ID obtained from the Salesforce Analytics Cloud system.

=== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | |x
| Data Set Id a| String | Identifier of a dataset within the Salesforce Analytics Cloud system. |  |x
| Records a| Array of Object | List of records to be inserted. |  `#[payload]` |

> | Target Variable a| String |  The variable name where the output of the operation is stored. |  |

Reviewer: ```suggestion
| Target Variable a| String |  The variable name where to store the operation's output. |  |
```

| Target Value a| String |  An expression to evaluate against the operation's output. The outcome of that expression is stored in the target variable. |  `#[payload]` |
| Reconnection Strategy a| * reconnect
* reconnect-forever |  A retry strategy in case of connectivity errors. |  |
|===

=== Output

[%header%autowidth.spread]
|===
| Type a| BulkOperationResult
|===

=== For Configurations

* salesforce-analytics

=== Throws

* SALESFORCE-ANALYTICS:CONNECTIVITY
* SALESFORCE-ANALYTICS:CONNECTIVITY
* SALESFORCE-ANALYTICS:INVALID_SESSION
* SALESFORCE-ANALYTICS:RETRY_EXHAUSTED
* SALESFORCE-ANALYTICS:TRANSACTION
* SALESFORCE-ANALYTICS:UNKNOWN

[[uploadExternalDataIntoNewDataSetAndStartProcessing]]
== Upload External Data Into New Dataset And Start Processing

`<salesforce-analytics:upload-external-data-into-new-data-set-and-start-processing>`

Creates a dataset, uploads data into it, and tells the Salesforce Analytics Cloud system to start processing the uploaded data.

=== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | |x

> | Type a| String |  Type of the records to be inserted. This is automatically computed based on metadata extracted from the provided file that contains the configuration element. |  |x

Reviewer: ```suggestion
| Type a| String |  Type of the records to insert. This automatically computes based on metadata extracted from the provided file that contains the configuration element. |  |x
```

| Records a| Array of Object |  List of records to insert. |  `#[payload]` |
| Operation a| Enumeration, one of:

** APPEND
** DELETE
** OVERWRITE
** UPSERT
|  |  |x
| Description a| String |  |  |x
| Label a| String |  |  |x
| Data Set Name a| String |  |  |x
| Edgemart Container a| String |  |  |
| Notification Sent a| Enumeration, one of:

** ALWAYS
** FAILURES
** NEVER
** WARNINGS
|  |  |
| Notification Email a| String |  |  |

> | Target Variable a| String |  The variable name where the output of the operation is stored. |  |

Reviewer: ```suggestion
| Target Variable a| String |  The variable name where to store the operation's output. |  |
```

* reconnect-forever |  A retry strategy in case of connectivity errors. |  |
|===

=== Output

[%header%autowidth.spread]
|===
| Type a| String
|===

=== For Configurations

* salesforce-analytics

=== Throws

* SALESFORCE-ANALYTICS:CONNECTIVITY
* SALESFORCE-ANALYTICS:CONNECTIVITY
* SALESFORCE-ANALYTICS:INVALID_SESSION
* SALESFORCE-ANALYTICS:RETRY_EXHAUSTED
* SALESFORCE-ANALYTICS:TRANSACTION
* SALESFORCE-ANALYTICS:UNKNOWN

== Types
[[Tls]]
=== Tls

[cols=".^20%,.^25%,.^30%,.^15%,.^10%", options="header"]
|======================
| Field | Type | Description | Default Value | Required
| Enabled Protocols a| String | A comma-separated list of protocols enabled for this context. |  |
| Enabled Cipher Suites a| String | A comma-separated list of cipher suites enabled for this context. |  |
| Trust Store a| <<TrustStore>> |  |  |
| Key Store a| <<KeyStore>> |  |  |
| Revocation Check a| * <<standard-revocation-check>>
* <<custom-ocsp-responder>>
* <<crl-file>> |  |  |
|======================

[[TrustStore]]
=== Trust Store

[cols=".^20%,.^25%,.^30%,.^15%,.^10%", options="header"]
|======================
| Field | Type | Description | Default Value | Required

> | Path a| String | The location (which will be resolved relative to the current classpath and file system, if possible) of the trust store. |  |

Reviewer: ```suggestion
| Path a| String | The location of the trust store, that if possible it will resolve relative to the current classpath and file system.. |  |
```

| Password a| String | The password used to protect the trust store. |  |
| Type a| String | The type of store used. |  |
| Algorithm a| String | The algorithm used by the trust store. |  |
| Insecure a| Boolean | If true, no certificate validations will be performed, rendering connections vulnerable to attacks. Use at your own risk. |  |
|======================

[[KeyStore]]
=== Key Store

[cols=".^20%,.^25%,.^30%,.^15%,.^10%", options="header"]
|======================
| Field | Type | Description | Default Value | Required

> | Path a| String | The location (which will be resolved relative to the current classpath and file system, if possible) of the key store. |  |

Reviewer: ```suggestion
| Path a| String | The location of the trust store, that if possible it will resolve relative to the current classpath and file system. |  |
```
